File: JavaScript/KthSmallestElementInABST/KthSmallestElementInABST.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * (158ms)
 * @param {TreeNode} root
 * @param {number} k
 * @return {number}
 */
var kthSmallest = function(root, k) {
    var count = [];
    dfs(root, count, k);
    return count[k-1];
};

var dfs = function(root, now, k) {
	if(root === null || now.length === k)
		return;
	dfs(root.left, now, k);
	now.push(root.val);
	dfs(root.right, now, k);
};

File: Python/KthSmallestElementInABST/KthSmallestElementInABST.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution(object):
	# (148ms)
    def kthSmallest(self, root, k):
        """
        :type root: TreeNode
        :type k: int
        :rtype: int
        """
        count = []
        self.dfs(root, count, k)
        return count[k - 1]

    def dfs(self, root, now, k):
    	if not root or len(now) == k:
    		return
    	self.dfs(root.left, now, k)
    	now.append(root.val)
    	self.dfs(root.right, now, k)